File: map/map.py
# Declaracion de un map (Diccionario)

# Representacion de un objeto
map = {
    "name": "Alvaro",
    "age": 25,
    "isAlive": True,
    "address": {
        "street": "Calle 123",
        "city": "Lima",
        "country": "Perú"
    }
}
print(map)

# Agregar
map["lastName"] = "Huaysara"

# Acceder
print(map["name"])
print(map["address"]["city"])

# Clave valor
reporte_x_anios = {
    2019: 100,
    2020: 200,
    2021: 300
}

print(reporte_x_anios)

# Claves
print(map.keys())
for key in map.keys():
    print(key)

# Valores
print(map.values())
for value in map.values():
    print(value)

# Items / Entries
print(map.items())
for key, value in map.items():
    print(f"Para la clave {key} se tiene el valor {value}") 

File: map/map.php
<?php

// Declaracion ded un map (array asociativo)
$map = array(
    "name" => "Alvaro",
    "age" => 25,
    "isAlive" => true,
    "address" => array(
        "street" => "Calle 123",
        "city" => "Lima",
        "country" => "Perú"
    ),
);

$map = [
    "name" => "Alvaro",
    "age" => 25,
    "isAlive" => true,
    "address" => [
        "street" => "Calle 123",
        "city" => "Lima",
        "country" => "Perú"
    ],
];

// Agregar
$map["lastName"] = "Huaysara";

// Acceso
print_r($map["name"]);
print_r($map["address"]["city"]);

// Claves
print_r(array_keys($map));
foreach (array_keys($map) as $key) {
    print_r($key);
}

// Valores
print_r(array_values($map));
foreach (array_values($map) as $value) {
    print_r($value);
}

// Items
print_r(array_map(null, array_keys($map), array_values($map)));
foreach (array_map(null, array_keys($map), array_values($map)) as [$key, $value]) {
    print_r("Para la clave $key se tiene el valor $value\n");
}
